Nykaa Marketing Campaign Performance Analysis Using SQL

Project Overview

This project analyzes Nykaa marketing campaign performance using SQL.
The objective is to evaluate campaign effectiveness, customer engagement,
conversion rates, ROI, and revenue generation across different marketing channels.

Import Required Libraries

In [2]:
import pandas as pd
import sqlite3

Load Dataset

In [3]:
df = pd.read_csv("nykaa_campaign_data.csv")

df.head()

,Campaign_ID,Campaign_Type,Target_Audience,Duration,Channel_Used,Impressions,Clicks,Leads,Conversions,Revenue,Acquisition_Cost,ROI,Language,Engagement_Score,Customer_Segment,Date
0,NY-CMP-1000,Social Media,College Students,21,"WhatsApp, YouTube",57804,6156,3616,2355,1867515,111.03,6.14,Hindi,20.98,College Students,29-04-2025
1,NY-CMP-1001,Paid Ads,Tier 2 City Customers,18,YouTube,91801,3321,1971,1357,1046247,180.83,3.26,Hindi,7.24,College Students,06-04-2025
2,NY-CMP-1002,Influencer,Youth,23,"WhatsApp, Google, YouTube",15536,2182,952,755,197055,90.60,1.88,English,25.03,College Students,14-01-2025
3,NY-CMP-1003,Email,Working Women,18,"YouTube, Facebook, Instagram",88114,8413,2231,947,376906,249.07,0.60,Hindi,13.15,College Students,04-06-2025
4,NY-CMP-1004,Paid Ads,College Students,10,"Facebook, Instagram",96871,3743,2060,1258,518296,228.60,0.80,Hindi,7.29,Tier 2 City Customers,29-12-2024


Basic Data Check

In [4]:
df.info()

df.isnull().sum()

df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55555 entries, 0 to 55554
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Campaign_ID       55555 non-null  object 
 1   Campaign_Type     55555 non-null  object 
 2   Target_Audience   55555 non-null  object 
 3   Duration          55555 non-null  int64  
 4   Channel_Used      55555 non-null  object 
 5   Impressions       55555 non-null  int64  
 6   Clicks            55555 non-null  int64  
 7   Leads             55555 non-null  int64  
 8   Conversions       55555 non-null  int64  
 9   Revenue           55555 non-null  int64  
 10  Acquisition_Cost  55555 non-null  float64
 11  ROI               55555 non-null  float64
 12  Language          55555 non-null  object 
 13  Engagement_Score  55555 non-null  float64
 14  Customer_Segment  55555 non-null  object 
 15  Date              55555 non-null  object 
dtypes: float64(3), int64(6), object(7)
memor

,Duration,Impressions,Clicks,Leads,Conversions,Revenue,Acquisition_Cost,ROI,Engagement_Score
count,55555.000000,55555.000000,55555.000000,55555.000000,55555.000000,5.555500e+04,55555.000000,55555.000000,55555.000000
mean,17.446044,55087.885357,4688.070507,1877.271119,1032.866925,5.158197e+05,377.347068,2.713807,13.784169
std,7.520438,25930.001514,3178.686285,1435.636117,862.496788,4.900121e+05,541.084524,4.493380,6.353125
min,5.000000,10001.000000,202.000000,56.000000,19.000000,6.183000e+03,9.080000,-0.970000,2.600000
25%,11.000000,32680.000000,2110.000000,779.000000,400.000000,1.777060e+05,105.435000,0.040000,8.360000
50%,17.000000,55182.000000,3907.000000,1481.000000,779.000000,3.604360e+05,207.510000,1.240000,13.600000
75%,24.000000,77514.500000,6688.000000,2605.000000,1414.000000,6.874225e+05,428.580000,3.630000,18.855000
max,30.000000,100000.000000,14868.000000,8876.000000,6686.000000,4.579910e+06,15473.160000,74.420000,30.910000


Create SQLite Database

In [5]:
conn = sqlite3.connect("NykaaMarketing.db")

cursor = conn.cursor()

Business Questions

1. What is the total revenue generated?
2. Which marketing channels drive the highest revenue?
3. Which campaign types have the highest ROI?
4. Which customer segments contribute most revenue?
5. Which target audiences convert best?
6. What are the top-performing campaigns?

Import Data into Table

In [6]:
df.to_sql(
    "marketing_campaign",
    conn,
    if_exists="replace",
    index=False
)

print("Data Imported Successfully")

Data Imported Successfully


Verify Data

In [7]:
query = """
SELECT *
FROM marketing_campaign
LIMIT 5
"""

pd.read_sql(query, conn)

,Campaign_ID,Campaign_Type,Target_Audience,Duration,Channel_Used,Impressions,Clicks,Leads,Conversions,Revenue,Acquisition_Cost,ROI,Language,Engagement_Score,Customer_Segment,Date
0,NY-CMP-1000,Social Media,College Students,21,"WhatsApp, YouTube",57804,6156,3616,2355,1867515,111.03,6.14,Hindi,20.98,College Students,29-04-2025
1,NY-CMP-1001,Paid Ads,Tier 2 City Customers,18,YouTube,91801,3321,1971,1357,1046247,180.83,3.26,Hindi,7.24,College Students,06-04-2025
2,NY-CMP-1002,Influencer,Youth,23,"WhatsApp, Google, YouTube",15536,2182,952,755,197055,90.60,1.88,English,25.03,College Students,14-01-2025
3,NY-CMP-1003,Email,Working Women,18,"YouTube, Facebook, Instagram",88114,8413,2231,947,376906,249.07,0.60,Hindi,13.15,College Students,04-06-2025
4,NY-CMP-1004,Paid Ads,College Students,10,"Facebook, Instagram",96871,3743,2060,1258,518296,228.60,0.80,Hindi,7.29,Tier 2 City Customers,29-12-2024


Total Revenue

In [8]:
query = """
SELECT
SUM(revenue) AS Total_Revenue
FROM marketing_campaign
"""

pd.read_sql(query, conn)

,Total_Revenue
0,28656364282


Total Campaign Spend

In [9]:
query = """
SELECT
SUM(acquisition_cost) AS Total_Spend
FROM marketing_campaign
"""

pd.read_sql(query, conn)

,Total_Spend
0,20963516.36


Overall ROAS

In [10]:
query = """
SELECT
ROUND(SUM(revenue)/SUM(acquisition_cost),2) AS Overall_ROAS
FROM marketing_campaign
"""

pd.read_sql(query, conn)

,Overall_ROAS
0,1366.96


Revenue by Channel

In [11]:
query = """
SELECT
channel_used,
SUM(revenue) AS Revenue
FROM marketing_campaign
GROUP BY channel_used
ORDER BY Revenue DESC
LIMIT 10
"""

pd.read_sql(query, conn)

,Channel_Used,Revenue
0,Instagram,1639493563
1,WhatsApp,1615380939
2,YouTube,1597611646
3,Email,1587734385
4,Google,1586013727
5,Facebook,1569289158
6,"WhatsApp, Google",358536574
7,"Email, YouTube",353128673
8,"WhatsApp, Facebook",334318714
9,"WhatsApp, Email",329884354


Best Campaign Type

In [12]:
query = """
SELECT
campaign_type,
SUM(revenue) AS Revenue
FROM marketing_campaign
GROUP BY campaign_type
ORDER BY Revenue DESC
"""

pd.read_sql(query, conn)

,Campaign_Type,Revenue
0,Influencer,5769064044
1,Social Media,5751837620
2,Paid Ads,5751468983
3,SEO,5698831847
4,Email,5685161788


Conversion Rate by Campaign Type

In [14]:
query = """
SELECT
campaign_type,
ROUND(
    SUM(conversions) * 100.0 / SUM(clicks),
    2
) AS Conversion_Rate
FROM marketing_campaign
GROUP BY campaign_type
ORDER BY Conversion_Rate DESC
"""

pd.read_sql(query, conn)

,Campaign_Type,Conversion_Rate
0,Email,22.12
1,Social Media,22.08
2,Influencer,22.05
3,SEO,22.03
4,Paid Ads,21.89


Top 10 Revenue Generating Campaigns

In [15]:
query = """
SELECT
campaign_id,
campaign_type,
channel_used,
revenue
FROM marketing_campaign
ORDER BY revenue DESC
LIMIT 10
"""

pd.read_sql(query, conn)

,Campaign_ID,Campaign_Type,Channel_Used,Revenue
0,NY-CMP-13007,Paid Ads,"Email, YouTube, Facebook",4579910
1,NY-CMP-44157,SEO,YouTube,4412709
2,NY-CMP-38379,Influencer,Instagram,4326120
3,NY-CMP-6741,Paid Ads,WhatsApp,4264178
4,NY-CMP-56543,Influencer,Facebook,4119654
5,NY-CMP-11367,Email,YouTube,4119352
6,NY-CMP-53612,Email,"Instagram, Email",4067481
7,NY-CMP-13914,Email,"WhatsApp, Instagram, Facebook",4066764
8,NY-CMP-40527,Email,"Email, YouTube",4022879
9,NY-CMP-3359,Email,"Google, WhatsApp",3891237


Customer Segment Analysis

In [16]:
query = """
SELECT
customer_segment,
SUM(revenue) AS Revenue,
SUM(conversions) AS Conversions,
ROUND(
SUM(revenue)*1.0/SUM(conversions),
2
) AS Revenue_Per_Conversion
FROM marketing_campaign
GROUP BY customer_segment
ORDER BY Revenue DESC
"""

pd.read_sql(query, conn)

,Customer_Segment,Revenue,Conversions,Revenue_Per_Conversion
0,Working Women,5811927015,11638803,499.36
1,College Students,5799502690,11583642,500.66
2,Premium Shoppers,5709910379,11445494,498.88
3,Tier 2 City Customers,5691846274,11365890,500.78
4,Youth,5643177924,11347093,497.32


Target Audience Analysis

In [17]:
query = """
SELECT
target_audience,
SUM(revenue) AS Revenue
FROM marketing_campaign
GROUP BY target_audience
ORDER BY Revenue DESC
"""

pd.read_sql(query, conn)

,Target_Audience,Revenue
0,Premium Shoppers,5852284945
1,College Students,5786247365
2,Working Women,5743709729
3,Tier 2 City Customers,5701063841
4,Youth,5573058402


Campaign Type ROI

In [18]:
query = """
SELECT
campaign_type,
ROUND(
SUM(revenue)/SUM(acquisition_cost),
2
) AS ROI
FROM marketing_campaign
GROUP BY campaign_type
ORDER BY ROI DESC
"""

pd.read_sql(query, conn)

,Campaign_Type,ROI
0,Social Media,1386.50
1,Influencer,1379.94
2,Paid Ads,1364.73
3,SEO,1358.14
4,Email,1345.92


Highest Conversion Campaign Types

In [19]:
query = """
SELECT
campaign_type,
SUM(conversions) AS Total_Conversions
FROM marketing_campaign
GROUP BY campaign_type
ORDER BY Total_Conversions DESC
"""

pd.read_sql(query, conn)

,Campaign_Type,Total_Conversions
0,Social Media,11595675
1,Influencer,11563919
2,Email,11425766
3,Paid Ads,11425573
4,SEO,11369989


Executive Summary

Key Findings

1. Total Revenue Generated: ₹28.66 Billion
2. Overall ROAS: 1367
3. Highest Revenue Channel: Instagram
4. Best Performing Campaign Type: Influencer
5. Highest ROI Campaign: Social Media
6. Best Customer Segment: Working Women

Business Recommendations

1. Increase budget allocation towards Instagram campaigns.
2. Expand Influencer Marketing partnerships.
3. Focus targeting on Working Women and Premium Shoppers.
4. Optimize Email campaigns for improved ROI.
5. Monitor monthly performance trends to identify underperforming periods.